### PR TITLE
[MINOR][DOCS] Tighten up some key links to the project and download pages to use HTTPS

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Tighten up some key links to the project and download pages to use HTTPS

## How was this patch tested?

N/A

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
LGTM. checked each link

**Review by felixcheung — APPROVED** [review]

File: R/WINDOWS.md [hunk 1]
```diff
@@ -20,19 +20,19 @@ license: |
 
 To build SparkR on Windows, the following steps are required
 
-1. Install R (>= 3.1) and [Rtools](http://cran.r-project.org/bin/windows/Rtools/). Make sure to
+1. Install R (>= 3.1) and [Rtools](https://cloud.r-project.org/bin/windows/Rtools/). Make sure to
```
Comment (dongjoon-hyun) [inline R/WINDOWS.md]: In this PR, this seems to be the only change to a better mirror site. +1.

**Review by dongjoon-hyun — APPROVED** [review]

File: R/WINDOWS.md (hunk 1 above)
Comment (srowen) [reply R/WINDOWS.md]: Yeah I could have included this with the other change too. This one is not a 'functional' change so I put it here.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

**Pull request closed without merging**